bundles/AdminBundle/Controller/Admin/AssetController.php

```diff
@@ -257,10 +257,8 @@ public function treeGetChildsByIdAction(Request $request, EventDispatcherInterfa
 
                 $condition = 'parentId = ' . $db->quote($asset->getId()) . ' and
                 (
-                (select list from users_workspaces_asset where userId in (' . implode(',', $userIds) . ') and LOCATE(CONCAT(path,filename),cpath)=1  ORDER BY LENGTH(cpath) DESC LIMIT 1)=1
-                or
-                (select list from users_workspaces_asset where userId in (' . implode(',', $userIds) . ') and LOCATE(cpath,CONCAT(path,filename))=1  ORDER BY LENGTH(cpath) DESC LIMIT 1)=1
-                    )';
+                    (SELECT list FROM users_workspaces_asset WHERE userId IN (' . implode(',', $userIds) . ') AND (LOCATE(CONCAT(path,filename),cpath)=1 OR LOCATE(cpath,CONCAT(path,filename))=1) ORDER BY LENGTH(cpath) DESC, FIELD(userId, ' . $this->getAdminUser()->getId() . ') DESC, list DESC LIMIT 1)=1
```

Comment:
merging sub queries has unexpected result now. 

when full permission is provided on root level for user:
![image](https://user-images.githubusercontent.com/5137917/86228378-9226fc00-bb8e-11ea-8d9b-3f53bae7de57.png)

and restriction is applied on sub level on user role
![image](https://user-images.githubusercontent.com/5137917/86228422-9fdc8180-bb8e-11ea-823b-863222d42832.png)

then it hides the complete root level for user.
![image](https://user-images.githubusercontent.com/5137917/86228584-d4e8d400-bb8e-11ea-9584-fa4be3744897.png)

Reply:
I don't think this is caused by the subquery merge.. can you confirm, that
```
$condition = 'parentId = ' . $db->quote($document->getId()) . ' and
                                        (
                                        (select list from users_workspaces_document where userId in (' . implode(',', $userIds) . ') and LOCATE(CONCAT(path,`key`),cpath)=1  ORDER BY LENGTH(cpath) DESC, FIELD(userId, 38) DESC, list LIMIT 1)=1
                                        or
                                        (select list from users_workspaces_document where userId in (' . implode(',', $userIds) . ') and LOCATE(cpath,CONCAT(path,`key`))=1  ORDER BY LENGTH(cpath) DESC, FIELD(userId, 38) DESC, list LIMIT 1)=1
                                        )';
```

for e.g. the DocumenController also returns 0 rows, though it should?

Reply:
I don't think this is caused by the merge of the subselect queries. Can you confirm, that
```
$condition = 'parentId = ' . $db->quote($document->getId()) . ' and
                                        (
                                        (select list from users_workspaces_document where userId in (' . implode(',', $userIds) . ') and LOCATE(CONCAT(path,`key`),cpath)=1  ORDER BY LENGTH(cpath) DESC, FIELD(userId, '. $this->getAdminUser()->getId() .') DESC, list DESC LIMIT 1)=1
                                        or
                                        (select list from users_workspaces_document where userId in (' . implode(',', $userIds) . ') and LOCATE(cpath,CONCAT(path,`key`))=1  ORDER BY LENGTH(cpath) DESC, FIELD(userId, '. $this->getAdminUser()->getId() .') DESC, list DESC LIMIT 1)=1
                                        )';
```
also returns 0 rows?

Reply:
yes, it returns expected rows. Could you please change it for all controllers? thanks!

Reply:
Of course...